src/calendarstorage.h

```diff
@@ -0,0 +1,279 @@
+/*
+  This file is part of the mkcal library.
+
+  Copyright (c) 2023 Damien Caliste <dcaliste@free.fr>
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Library General Public
+  License as published by the Free Software Foundation; either
+  version 2 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Library General Public License for more details.
+
+  You should have received a copy of the GNU Library General Public License
+  along with this library; see the file COPYING.LIB.  If not, write to
+  the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+  Boston, MA 02110-1301, USA.
+*/
+/**
+  @file
+  This file is part of the API for handling calendar data and
+  defines the an interface to store a single KCalendarCore::Calendar.
+
+  @author Damien Caliste \<dcaliste@free.fr\>
+*/
+
+#ifndef MKCAL_CALENDARSTORAGE_H
+#define MKCAL_CALENDARSTORAGE_H
+
+#include "mkcal_export.h"
+#include "notebook.h"
+
+#include <KCalendarCore/CalStorage>
+#include <KCalendarCore/MemoryCalendar>
+
+namespace KCalendarCore {
+class Incidence;
+}
+
+namespace mKCal {
+
+/**
+  @brief
+  This class provides a calendar storage interface.
+  Every action on the storage can be synchronous or asynchronous,
+  depending on the storage implementation. SqliteStorage is a
+  synchronous implementation. 
+
+  In any case, caller can use CalendarStorage::Observer to get
+  notified about the action done.
+*/
+class MKCAL_EXPORT CalendarStorage
+    : public KCalendarCore::CalStorage
+{
+    Q_OBJECT
+
+public:
+
+    /**
+      Action to be performed on save for deleted incidences.
+    */
+    enum DeleteAction {
```

Comment:
It will disappear in ExtendedStorage with it. So Remains in CalendarStorage and MultiCalendarStorage. If I look at the API that is common between the two, I only find this enum. Even the save() routine that is using it has a different API. So I'm not sure it's worth trying to abstract something in common between the two classes.